relay-server/src/utils/rate_limits.rs

```diff
@@ -666,7 +677,12 @@ where
             rate_limits.merge(checkin_limits);
         }
 
-        if summary.span_quantity > 0 {
+        // We want to process spans rate limits only if they were not already applied because the
+        // transaction had child spans.
+        if summary.span_quantity > 0
+            && enforcement.span_metrics.is_default()
+            && enforcement.spans.is_default()
```

Comment:
The rationale behind this condition is that, if we have already applied span enforcements, we don't want to try and fetch quotas because:
1. We will waste compute
2. We will override the enforcement with default in case the `check` doesn't succeed